### PR TITLE
17.0 mig web tree many2one clickable mwi

### PR DESCRIPTION
**Pull request by maciej-wichowski**

### CODE REVIEW

File: web_tree_many2one_clickable/static/src/components/many2one_button/many2one_button.esm.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            res_id: this.props.value[0],
+            views: [[false, "form"]],
+            target: "target",
+            additionalContext: this.props.context || {},
+        });
+    }
+}
+TreeMany2oneClickableButton.template = "web_tree_many2one_clickable.Button";
+
+Object.assign(ListRenderer.components, {TreeMany2oneClickableButton});
```
Comment (oerp-odoo) [inline web_tree_many2one_clickable/static/src/components/many2one_button/many2one_button.esm.js:34]: Was this done originally? Shouldn't we use "patch" to modify existing component?

**Review by oerp-odoo — APPROVED** [review]

File: web_tree_many2one_clickable/static/src/components/many2one_button/many2one_button.esm.js (hunk 1 above)
Comment (maciej-wichowski) [reply web_tree_many2one_clickable/static/src/components/many2one_button/many2one_button.esm.js:34]: It is a part of the code included in migration to v16. I will investigate if we can use `patch`

Comment (maciej-wichowski) [reply web_tree_many2one_clickable/static/src/components/many2one_button/many2one_button.esm.js:34]: @oerp-odoo I couldn't find any examples of extending components of existing class via `patch`. But `Object.assign` is quite commonly used [[example](https://github.com/versada/odoo/blob/9dba4e0d77187f0158243d314057f7aa3a19b543/addons/mail/static/src/discuss/call/common/chat_window_patch.js#L6)]

Comment (dz0) [reply web_tree_many2one_clickable/static/src/components/many2one_button/many2one_button.esm.js:34]: suggest not to change if it works,  but opportunity to  try understand js specifics:

docs have example for [static properties](https://www.odoo.com/documentation/17.0/developer/reference/frontend/patching_code.html?highlight=static%20properties#patching-a-javascript-class), 

But [js static properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static) are not propagated to instances (like in python)..

Still class methods can be patched via `prototype` (in same docs), maybe `components` property could also...

**Review by dz0 — APPROVED** [review]

**Pull request merged**